src/libraries/System.Diagnostics.Process/tests/ProcessStartInfoTests.cs

```diff
@@ -1347,5 +1317,326 @@ private void VerifyNotepadMainWindowTitle(Process process, string filename)
+            }
+            catch (Win32Exception ex) when (ex.NativeErrorCode == ERROR_SHARING_VIOLATION)
+            {
+                throw new SkipTestException($"{process.StartInfo.FileName} has been locked by some other process");
+            }
+        }
+
+        private class TestProcessState : IDisposable
+        {
+            private readonly Process _process;
+
+            private readonly bool _hasStarted;
+
+            private readonly WindowsTestAccount _processAccount;
+
+            private readonly WindowsTestAccount _processImpersonationAccount;
+
+            private readonly string _workingDirectory;
+
+            private readonly Action<string, string> _additionalCleanup;
+
+            private bool _disposedValue;
+
+            public TestProcessState(
+                Process process,
+                bool hasStarted,
+                WindowsTestAccount processAccount,
+                WindowsTestAccount processImpersonationAccount,
+                string workingDirectory,
+                Action<string, string> additionalCleanup)
+            {
+                _process = process;
+                _hasStarted = hasStarted;
+                _processAccount = processAccount;
+                _processImpersonationAccount = processImpersonationAccount;
+                _workingDirectory = workingDirectory;
+                _additionalCleanup = additionalCleanup;
+            }
+
+            public string ProcessAccountName => _processAccount?.AccountName;
+
+            public string ImpersonationAccountName => _processImpersonationAccount?.AccountName;
+
+            protected virtual void Dispose(bool disposing)
+            {
+                if (!_disposedValue)
+                {
+                    if (disposing)
+                    {
+                        if (_hasStarted)
+                        {
+                            _process.Kill();
+
+                            Assert.True(_process.WaitForExit(WaitInMS));
+                        }
+
+                        if (PlatformDetection.IsNotWindowsServerCore)
+                        {
+                            // remove the access
+                            SetAccessControl((_processImpersonationAccount ?? _processAccount).AccountName, _process.StartInfo.FileName, _workingDirectory, add: false);
+                        }
+
+                        _additionalCleanup?.Invoke(_processAccount?.AccountName, _processImpersonationAccount?.AccountName);
+                        _processAccount?.Dispose();
+                        _processImpersonationAccount?.Dispose();
+                    }
+
+                    _disposedValue = true;
+                }
+            }
+
+            public void Dispose()
+            {
+                Dispose(disposing: true);
+                GC.SuppressFinalize(this);
+            }
+        }
+    }
+
+    public partial class TestFileShare : IDisposable
```

Comment:
@adamsitnik I took this code from https://github.com/dotnet/runtime/blob/b0920912512aa7e205363b046a1911e6a97df31f/src/libraries/System.IO.FileSystem/tests/FileStream/FileStreamConformanceTests.Windows.cs#L108-L161 and modified it a bit to be a single, disposable file share. Should I move this to a separate class in `Common/tests/TestUtilities` (same level as `WindowsTestAccount`) and refactor `FileStreamConformanceTests` and this test to use it? Or would this increase the scope of this PR too much, since I'm not really touching `System.IO.FileSystem` files for this PR otherwise? Not sure about the usual policies for this repo. :)

Reply:
> Should I move this to a separate class in Common/tests/TestUtilities (same level as WindowsTestAccount) and refactor FileStreamConformanceTests and this test to use it?

That would be great! 🥇 

> Or would this increase the scope of this PR too much, since I'm not really touching System.IO.FileSystem files for this PR otherwise? Not sure about the usual policies for this repo. :)

I currently own `System.IO.FileSystem` and I am fine with that ;)

Reply:
eef186a90d2cad6e87808ed56f468c5dfeb0f86d